packages/jest-circus/src/eventHandler.ts

```diff
@@ -147,7 +149,29 @@ const eventHandler: EventHandler = (event, state): void => {
       break;
     }
     case 'test_retry': {
+      // Clear errors so tests can be retried (and not immediately fail)
       event.test.errors = [];
+
+      // Clear any snapshot data that occurred in previous test run
+      const tmpSnapshotState = global.expect.getState().snapshotState;
```

Comment:
I think it would make sense for `global.expect` to expose `clearState`, call it here and let it do the job of cleaning up

Reply:
Totally agree. I should have some time this week to look at this some more.